docusaurus/docs/how-to-guides/panel-plugins/plugin-internationalization.md

```diff
@@ -0,0 +1,306 @@
+---
+id: plugin-internationalization
+title: How to add translations to your plugin
+description: Translate your plugins
+keywords:
+  - grafana
+  - plugins
+  - plugin
+  - frontend
+  - tooling
+  - configuration
+  - translation
+  - localization
+  - internationalization
+  - webpack
+---
+
+# How to add translations to your plugin
+
+The Grafana UI is available in [several languages](https://grafana.com/docs/grafana/latest/administration/organization-preferences/#change-grafana-language). By default, plugins are available in English only and are not translated when you change your language settings. 
+
+Grafana uses the [i18next internationalization framework](https://www.i18next.com/) for translation purposes. If you want your plugin to be translatable to other languages you need to perform the changes described in this document.  
+
+## Before you begin
+
+:::info
+Available in Grafana >=v11.0.0 (US English) and Grafana >=v12.1.0 for the rest of languages.
+:::
+
+The following is recommended:
+
+* Basic knowledge of Grafana panel plugin development
+* Basic understanding of the [`t` function](https://www.i18next.com/overview/api#t)
+* Basic understanding of the [`Trans` component](https://react.i18next.com/latest/trans-component)
+
+## Overview of the files affected by translation
+
+If you create your plugin running the `create-plugin` scaffolding tool, enabling plugin translation involves updating the following files:
+
+- `docker-compose.yaml` 
+- `plugin.json`
+- `module.ts`
+- `webpack.config.ts`
+- `package.json`
+
+By the end of the translation process you'll have a file structure like this:
+
+```
+myorg-myplugin-plugintype/
+├── pkg/
+│   ├── main.go
+│   └── plugin/
+├── src/
+│   ├── locales
+│      ├── en-US
+│         └──--.json 
```

Comment:
Don't these json files have names based on the plugin-id?
```suggestion
│         └──myorg-myplugin-plugintype.json 
```

Reply:
Good catch 🎉